File: 2022/06/19/java/SummaryRanges.java
/**
 * https://leetcode.com/problems/summary-ranges/
 * @author Andrew Jarombek
 * @since 6/19/2022
 */

import java.util.ArrayList;
import java.util.List;

public class SummaryRanges {
    public List<String> summaryRanges(int[] nums) {
        List<String> result = new ArrayList<>();

        if (nums.length == 0)
            return result;

        int start = nums[0];
        int current = start;

        for (int i = 1; i < nums.length; i++) {
            int val = nums[i];

            if (val - 1 == current) {
                current = val;
            } else {
                if (start == current) {
                    result.add(String.valueOf(start));
                } else {
                    result.add(start + "->" + current);
                }

                start = val;
                current = val;
            }
        }

        if (start == current) {
            result.add(String.valueOf(start));
        } else {
            result.add(start + "->" + current);
        }

        return result;
    }
}

File: 2022/11/09/go/summary_ranges.go
/**
 * https://leetcode.com/problems/power-of-two/
 * Author: Andrew Jarombek
 * Date: 11/9/2022
 */

package _go

func isPowerOfTwo(n int) bool {
	if n < 1 {
		return false
	}

	for n > 1 {
		if n%2 != 0 {
			return false
		}

		n = n / 2
	}

	return true
}
